libraries/AP_Notify/ToneAlarm_PX4.cpp

```diff
@@ -162,6 +164,14 @@ void ToneAlarm_PX4::update()
     }
 
     check_cont_tone();
+    
+    if (AP_Notify::flags.powering_off) {
+        if (!flags.powering_off) {
+            play_tone(AP_NOTIFY_PX4_TONE_QUIET_SHUTDOWN);
+        }
+        flags.powering_off = AP_Notify::flags.powering_off;
```

Comment:
I'm not sure what you mean? The `AP_Notify::flags.powering_off` flag will be false unless the user pushes the power button fox 2 seconds, at which point it goes true. The flags internal to the tone file are just to keep it from repeating once it's fired once.  Otherwise it be playing the tone at 50hz.